Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -838,6 +1091,290 @@ void MatchTOF::doMatching(int sec)
+        deltaPos[imatch][0] /= nStepsInsideSameStrip[imatch];
+        deltaPos[imatch][1] /= nStepsInsideSameStrip[imatch];
+        deltaPos[imatch][2] /= nStepsInsideSameStrip[imatch];
+        //   LOG(DEBUG) << "matched strip " << imatch << ": deltaPos[0] = " << deltaPos[imatch][0] << ", deltaPos[1] = " << deltaPos[imatch][1] << ", deltaPos[2] = " << deltaPos[imatch][2] << ", residual (x, z) = " << TMath::Sqrt(deltaPos[imatch][0] * deltaPos[imatch][0] + deltaPos[imatch][2] * deltaPos[imatch][2]);
+      }
+
+      if (nStripsCrossedInPropagation == 0) {
+        continue; // the track never hit a TOF strip during the propagation
+      }
+      bool foundCluster = false;
+      itof0 = 0;
+      for (auto itof = itof0; itof < nTOFCls; itof++) {
+        //      printf("itof = %d\n", itof);
+        auto& trefTOF = mTOFClusWork[cacheTOF[itof]];
+        // compare the times of the track and the TOF clusters - remember that they both are ordered in time!
+        //Printf("trefTOF.getTime() = %f, maxTrkTime = %f, minTrkTime = %f", trefTOF.getTime(), maxTrkTime, minTrkTime);
+
+        if (trefTOF.getTime() < minTrkTime) { // this cluster has a time that is too small for the current track, we will get to the next one
+          //Printf("In trefTOF.getTime() < minTrkTime");
+          itof0 = itof + 1; // but for the next track that we will check, we will ignore this cluster (the time is anyway too small)
+          continue;
+        }
+        if (trefTOF.getTime() > maxTrkTime) { // no more TOF clusters can be matched to this track
+          break;
+        }
+
+        int mainChannel = trefTOF.getMainContributingChannel();
+        int indices[5];
+        Geo::getVolumeIndices(mainChannel, indices);
+
+        // compute fine correction using cluster position instead of pad center
+        // this because in case of multiple-hit cluster position is averaged on all pads contributing to the cluster (then error position matrix can be used for Chi2 if nedeed)
+        int ndigits = 1;
+        float posCorr[3] = {0, 0, 0};
+
+        if (trefTOF.isBitSet(Cluster::kLeft))
+          posCorr[0] += Geo::XPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kUpLeft))
+          posCorr[0] += Geo::XPAD, posCorr[2] -= Geo::ZPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kDownLeft))
+          posCorr[0] += Geo::XPAD, posCorr[2] += Geo::ZPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kUp))
+          posCorr[2] -= Geo::ZPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kDown))
+          posCorr[2] += Geo::ZPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kRight))
+          posCorr[0] -= Geo::XPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kUpRight))
+          posCorr[0] -= Geo::XPAD, posCorr[2] -= Geo::ZPAD, ndigits++;
+        if (trefTOF.isBitSet(Cluster::kDownRight))
+          posCorr[0] -= Geo::XPAD, posCorr[2] += Geo::ZPAD, ndigits++;
+
+        if (ndigits > 1) {
+          posCorr[0] /= ndigits;
```

Comment:
better do once ``float ndifInv = 1./ndigits;`` then multiply by it.

Reply:
done